feature-libs/cart/saved-cart/components/saved-cart-form-dialog/saved-cart-form-dialog.component.spec.ts

```diff
@@ -80,5 +113,128 @@ describe('SavedCartFormDialogComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  // TODO: michal to add unit tests
+  describe('should build the form', () => {
+    it('with fetched cart values', () => {
+      mockDialogData$.next(mockFilledDialogData);
+      fixture.detectChanges();
+
+      expect(component.form?.get('name')?.value).toEqual(mockCart.name);
+      expect(component.form?.get('description')?.value).toEqual(
+        mockCart.description
+      );
+    });
+
+    it('with empty values', () => {
+      mockDialogData$.next(mockEmptyDialogData);
+      fixture.detectChanges();
+
+      expect(component.form?.get('name')?.value).toEqual('');
+      expect(component.form?.get('description')?.value).toEqual('');
+    });
+  });
+
+  describe('should call service on save cart', () => {
+    beforeEach(() => {
+      spyOn(savedCartService, 'saveCart');
+    });
+
+    it('with filled values', () => {
+      mockDialogData$.next(mockFilledDialogData);
+      fixture.detectChanges();
+      component.saveOrEditCart(mockCartId);
+
+      expect(savedCartService.saveCart).toHaveBeenCalledWith({
+        cartId: mockCartId,
+        saveCartName: mockCart.name,
+        saveCartDescription: mockCart.description,
+        extraData: { edit: true },
+      });
+    });
+
+    it('with empty values', () => {
+      mockDialogData$.next(mockEmptyDialogData);
+      fixture.detectChanges();
+      component.saveOrEditCart(mockCartId);
+
+      expect(savedCartService.saveCart).toHaveBeenCalledWith({
+        cartId: mockCartId,
+        saveCartName: '',
+        saveCartDescription: '',
+        extraData: { edit: false },
+      });
+    });
+  });
+
+  describe('on delete cart', () => {
```

Comment:
You can just put them all in one `it`, which calls the deleteCart and expects those things to get triggered